r/R/dataset-format.R

```diff
@@ -118,102 +118,71 @@ IpcFileFormat <- R6Class("IpcFileFormat", inherit = FileFormat)
 #' @rdname FileFormat
 #' @export
 CsvFileFormat <- R6Class("CsvFileFormat", inherit = FileFormat)
-CsvFileFormat$create <- function(...,
-                                 opts = csv_file_format_parse_options(...),
-                                 convert_options = csv_file_format_convert_opts(...),
-                                 read_options = csv_file_format_read_opts(...)) {
-  check_csv_file_format_args(...)
-  # Evaluate opts first to catch any unsupported arguments
-  force(opts)
-
-  options <- list(...)
-  schema <- options[["schema"]]
-  if (!is.null(schema) && !inherits(schema, "Schema")) {
-    abort(paste0(
-      "`schema` must be an object of class 'Schema' not '",
-      class(schema)[1],
-      "'."
-    ))
-  }
+CsvFileFormat$create <- function(...) {
```

Comment:
whereby:

- AFAIK this isn't intended to be used much by users directly,
- but it is exported, so at least some power users probably will anyway, 
- and I appreciate the flattening here, 
- but whenever more parameters get shoved through `...` it gets harder for users to figure out what they can specify there (and specifically here you'll have to dig through more levels of source to figure it out),

suggested:

- could we document where to look in `?FileFormat` a bit more?
- could that page specify that`CsvFileFormat` exists if this is going to `@rdname` there?

Reply:
Good points made; the refactoring I did here is because that bit of the code was overly complex, so in the name of usability, I think it'll make more sense to write actual independent docs for this function with more explicit information about its usage.